app/controllers/merchant/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Merchant::BaseController < ApplicationController
+  before_action :require_merchant
+
+  def require_merchant
```

Comment:
Add before actions in Merchant base controller to limit who can see merchant dashboard